File: Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/AppopenFragment.java
package com.vungle.samples.samplejava.ui.fragment;

import androidx.annotation.NonNull;
import com.vungle.ads.AdConfig;
import com.vungle.ads.BaseAd;
import com.vungle.ads.InterstitialAd;
import com.vungle.ads.InterstitialAdListener;
import com.vungle.ads.VungleError;
import com.vungle.samples.samplejava.R;

public class AppopenFragment extends AdExperienceFragment implements InterstitialAdListener {

    private InterstitialAd interstitialAd;

    public AppopenFragment() {
        // doesn't do anything special
    }

    @Override
    protected void playAd() {
        super.playAd();
        if (interstitialAd != null && interstitialAd.canPlayAd()) {
            interstitialAd.play(requireContext());
        }
    }

    @Override
    protected void loadAd() {
        super.loadAd();
        interstitialAd = new InterstitialAd(requireActivity(), placementId, new AdConfig());
        interstitialAd.setAdListener(this);
        interstitialAd.load("");
    }

    @Override
    public void onAdClicked(@NonNull BaseAd baseAd) {
        setTextColor(_binding.lbAdClick, R.color.black);
    }

    @Override
    public void onAdEnd(@NonNull BaseAd baseAd) {
        setTextColor(_binding.lbAdEnd, R.color.black);
    }

    @Override
    public void onAdFailedToLoad(@NonNull BaseAd baseAd, @NonNull VungleError vungleError) {
        setTextColor(_binding.lbError, R.color.black);
        _binding.lbError.setText(vungleError.getErrorMessage());
    }

    @Override
    public void onAdFailedToPlay(@NonNull BaseAd baseAd, @NonNull VungleError vungleError) {
        setTextColor(_binding.lbError, R.color.black);
        _binding.lbError.setText(vungleError.getErrorMessage());
    }

    @Override
    public void onAdImpression(@NonNull BaseAd baseAd) {
        setTextColor(_binding.lbAdImpression, R.color.black);
    }

    @Override
    public void onAdLeftApplication(@NonNull BaseAd baseAd) {
        setTextColor(_binding.lbOnAdLeftApplication, R.color.black);
    }

    @Override
    public void onAdLoaded(@NonNull BaseAd baseAd) {
        setTextColor(_binding.lbAdLoaded, R.color.black);
        _binding.btnPlayAd.setEnabled(true);
        setTextColor(_binding.btnPlayAd, R.color.black);
    }

    @Override
    public void onAdStart(@NonNull BaseAd baseAd) {
        setTextColor(_binding.lbAdStart, R.color.black);
        _binding.btnPlayAd.setEnabled(false);
        setTextColor(_binding.btnPlayAd, R.color.lightGray);
    }
}

File: Samples/android/Kotlin/app-v7-kotlin/src/main/java/com/vungle/samples/samplekotlin/AppopenFragment.kt
package com.vungle.samples.samplekotlin

import android.os.Bundle
import android.view.View
import com.vungle.samples.samplekotlin.ui.AdExperienceFragment
import com.vungle.ads.AdConfig
import com.vungle.ads.BaseAd
import com.vungle.ads.InterstitialAd
import com.vungle.ads.InterstitialAdListener
import com.vungle.ads.VungleError

class AppopenFragment : AdExperienceFragment(), InterstitialAdListener {

  private var interstitialAd: InterstitialAd? = null

  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)

    configureUi()
  }

  override fun loadAd() {
    super.loadAd()

    interstitialAd = InterstitialAd(requireContext(), placementId, AdConfig().apply {
      adOrientation = AdConfig.AUTO_ROTATE
    }).apply {
      adListener = this@AppopenFragment
      load()
    }
  }

  override fun onAdLoaded(baseAd: BaseAd) {
    super.onAdLoaded(baseAd)
  }

  override fun onAdStart(baseAd: BaseAd) {
    super.onAdStart(baseAd)
  }

  override fun onAdImpression(baseAd: BaseAd) {
    super.onAdImpression(baseAd)
  }

  override fun onAdEnd(baseAd: BaseAd) {
    super.onAdEnd(baseAd)
  }

  override fun onAdClicked(baseAd: BaseAd) {
    super.onAdClicked(baseAd)
  }

  override fun onAdLeftApplication(baseAd: BaseAd) {
    super.onAdLeftApplication(baseAd)
  }

  override fun onAdFailedToLoad(baseAd: BaseAd, adError: VungleError) {
    super.onAdFailedToLoad(baseAd, adError)
  }

  override fun onAdFailedToPlay(baseAd: BaseAd, adError: VungleError) {
    super.onAdFailedToPlay(baseAd, adError)
  }

  override fun playAd() {
    super.playAd()
    if (interstitialAd?.canPlayAd() == true) {
      interstitialAd?.play(requireContext())
    }
  }


  private fun configureUi() {
    binding.lbPlacementId.text = placementId
  }

  override fun onDestroyView() {
    super.onDestroyView()
    interstitialAd?.adListener = null
    interstitialAd = null
  }
}